ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/engine/UtilsTest.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.tests.engine
+
+import io.ktor.client.engine.*
+import io.ktor.client.utils.*
+import io.ktor.http.*
+import kotlin.test.*
+
+class UtilsTest {
+    @Test
+    fun testMergeHeaders() {
```

Comment:
Could you add an end-to-end test with a socket server for all engines as well?

Reply:
I added several tests, please have a look.